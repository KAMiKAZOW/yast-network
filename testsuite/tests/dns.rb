# encoding: utf-8

module Yast
  class DnsClient < Client
    def main
      Yast.import "Assert"
      Yast.import "Testsuite"

      @READ = {
        "probe"     => { "architecture" => "i386" },
        "sysconfig" => { "console" => { "CONSOLE_ENCODING" => "UTF-8" } }
      }

      @EXEC = {
        "target" => {
          "bash_output" => {
            "exit"   => 0,
            "stdout" => "charset=UTF-8",
            "stderr" => ""
          }
        }
      }

      @hostnames = [
        "127.0.0.1       localhost",
        "1.1.1.1         1-hostname alias",
        "2.2.2.2         hostname-2",
        "3.3.3.3         -invalid_hostname-",
        "9.114.214.42    9117-mma-1-lp1.pok.stglabs.ibm.com"
      ]

      Yast.import "DNS"

      Assert.Equal(
        "localhost",
        DNS.GetHostnameFromGetent(Ops.get(@hostnames, 0, ""))
      )
      Assert.Equal(
        "1-hostname",
        DNS.GetHostnameFromGetent(Ops.get(@hostnames, 1, ""))
      )
      Assert.Equal(
        "hostname-2",
        DNS.GetHostnameFromGetent(Ops.get(@hostnames, 2, ""))
      )
      Assert.Equal("", DNS.GetHostnameFromGetent(Ops.get(@hostnames, 3, "")))
      Assert.Equal(
        "9117-mma-1-lp1.pok.stglabs.ibm.com",
        DNS.GetHostnameFromGetent(Ops.get(@hostnames, 4, ""))
      )

      nil
    end
  end
end

Yast::DnsClient.new.main
