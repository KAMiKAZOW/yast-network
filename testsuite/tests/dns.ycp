{

include "testsuite.ycp";
include "network/dns.ycp";

DUMP("check_hostname");
TEST(``(check_hostname(nil)), [], nil);
TEST(``(check_hostname("")), [], nil);
TEST(``(check_hostname("A_B87126")), [], nil);
TEST(``(check_hostname("ahoj.joha")), [], nil);
TEST(``(check_hostname("ahoj joha")), [], nil);
TEST(``(check_hostname("ABC-012345678901234567890123456789012345678901234567890123456789")), [], nil);
TEST(``(check_hostname("----------")), [], nil);
DUMP("----------");
TEST(``(check_hostname("8abc")), [], nil);
TEST(``(check_hostname("ahoj")), [], nil);
TEST(``(check_hostname("aHoJ")), [], nil);
TEST(``(check_hostname("A-B87126")), [], nil);
TEST(``(check_hostname("A0123456789")), [], nil);
TEST(``(check_hostname("AB-012345678901234567890123456789012345678901234567890123456789")), [], nil);
TEST(``(check_hostname("abc-")), [], nil);
TEST(``(check_hostname("123")), [], nil);

DUMP("check_fqhostname");
TEST(``(check_fqhostname("www.blah.com")), [], nil);
TEST(``(check_fqhostname("123.com")), [], nil);

DUMP("check_domainname");
TEST(``(check_domainname(nil)), [], nil);
TEST(``(check_domainname("")), [], nil);
TEST(``(check_domainname("ww.asda.")), [], nil);
TEST(``(check_domainname(".asd.asd.com")), [], nil);
TEST(``(check_domainname("www.a sd.com")), [], nil);
TEST(``(check_domainname("dsf.234")), [], nil);
TEST(``(check_domainname("dsf.4com")), [], nil);
TEST(``(check_domainname("asd-.com")), [], nil);
TEST(``(check_domainname("----------")), [], nil);
DUMP("----------");
TEST(``(check_domainname("A.B")), [], nil);
TEST(``(check_domainname("www.blah.com")), [], nil);
TEST(``(check_domainname("as.cca.da3.cdd222.cd-2s.com")), [], nil);
TEST(``(check_domainname("www.a-sd.com")), [], nil);
TEST(``(check_domainname("asdf.2234-dsf.com")), [], nil);
TEST(``(check_domainname("12-34.56-78.com")), [], nil);
TEST(``(check_domainname("123.com")), [], nil);
TEST(``(check_domainname("4com")), [], nil);
TEST(``(check_domainname("doncom")), [], nil);

DUMP("split_fqhostname");
TEST(``(split_fqhostname("ftp.suse.cz")), [], nil);
TEST(``(split_fqhostname("123.com")), [], nil);
TEST(``(split_fqhostname("beholder")), [], nil);

}
