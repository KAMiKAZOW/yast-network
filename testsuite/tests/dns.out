Dump	check_hostname
Return	false
Return	false
Return	false
Return	false
Return	false
Return	false
Return	false
Dump	----------
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	false
Return	true
Dump	check_fqhostname
Return	true
Return	true
Dump	check_domainname
Return	false
Return	false
Return	false
Return	false
Return	false
Return	false
Return	false
Return	false
Return	false
Dump	----------
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Return	true
Dump	split_fqhostname
Return	["ftp", "suse.cz"]
Return	["123", "com"]
Return	["beholder"]
